File: POBJ/CPP/Exercices/EX_4/EX_4/TriangleRectangle.cpp
#include "TriangleRectangle.h"
#include <cmath>


// Constructeur
TriangleRectangle :: TriangleRectangle()
{
    std::cout << "Constructeur du TriangleRectangle" << std::endl;
}

double TriangleRectangle::CalculerSurface(double base, double hauteur)
{
    Surface = 0.5 * base * hauteur;
    return Surface;  // Formule pour la surface d'un triangle rectangle (base * hauteur / 2), mais la division dans le code est moins bien que les multiplications 
}

double TriangleRectangle::CalculerPerimetre(double base, double hauteur)
{
    // Pour un triangle rectangle, il faudra également calculer l'hypoténuse 
    double hypotenuse = sqrt(base * base + hauteur * hauteur);
    Perimetre = base + hauteur + hypotenuse;
    return Perimetre; // Formule pour le périmètre d'un triangle rectangle
}

File: POBJ/Csharp/Ex_4_Reprise_Cpp/Ex_4_Reprise_Cpp/TriangleRectangle.cs
using System;

public class TriangleRectangle : FormesGeometriques
{
    // Constructeur
    public TriangleRectangle()
    {
        Console.WriteLine("Constructeur du TriangleRectangle");
    }

    public override double CalculerSurface(double baseTri, double hauteurTri)
    {
        Surface = 0.5 * baseTri * hauteurTri;
        return Surface;  // Formule pour la surface d'un triangle rectangle (base * hauteur / 2), mais la division dans le code est moins bien que les multiplications 
    }

    public override double CalculerPerimetre(double baseTri, double hauteurTri)
    {
        // Pour un triangle rectangle, il faudra également calculer l'hypoténuse 
        double hypotenuse = Math.Sqrt(baseTri * baseTri + hauteurTri * hauteurTri);
        Perimetre = baseTri + hauteurTri + hypotenuse;
        return Perimetre; // Formule pour le périmètre d'un triangle rectangle
    }
}
